### PR TITLE
Extend DetailTop component to have icon and tooltip

### LINKED ISSUE #7597: Extend DetailTop component to have icon and tooltip
# Description

As outcome of https://github.com/rancher/dashboard/issues/7119, add ability to:
- Display icons
- Replace key with value and display key in tooltips over tabs.

## Context

From a given dictionary of key terms, which are the values displayed on the tab, map the tooltips.
Add conditionally icons on the left from a give string.

Story for the style guide needs to be added for this component.

Design not provided.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #7597
As defined in the issue.

### Occurred changes and/or fixed issues
- Added icons dictionary, e.g. `{ 'labelKey': 'icon-rancher-icon' }`
- Added tooltips dictionary to replace content, e.g. `{ 'labelKey': 'literaly any type of text' }`

### Technical notes summary
- Created story for the component

### Screenshot/Video

Screenshot 2022-12-01 at 15 04 29

### CODE REVIEW

File: stories/components/DetailTop.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  light
+  format={false}
+  code={`
+     import DetailTop from '@shell/components/DetailTop.vue'
+  `}
+/>
+
+### Props table
+
+<ArgsTable of={Banner} />
```
Comment (torchiaf) [inline stories/components/DetailTop.stories.mdx:96]: I think this should be `<ArgsTable of={DetailTop} />`

**Review by torchiaf — CHANGES_REQUESTED** [review]
There is a bug.

To reproduce:

1. Go to the storybook, DetailTop -> Tooltips
2. put the mouse pointer over the label to display the tooltip value

The tooltip value seems to be that one in the `value.labels` prop, not the one of `tooltips` prop, please see the image below:

image

**Comment by cnotv** [comment]
@torchiaf the tooltip should replace the `key : value` label.
The value is not present in the tooltip dictionary in any form, so it's not possible to modify the tooltip from it as it's just bound to it.
I am adding some informations to it, as the logic does not seems straightforward.

**Review by torchiaf — APPROVED** [review]
LGTM, let's investigate the tooltip behavior separately

**Pull request merged**